### PR TITLE
Hide settings in site-only conn for non admin

### PR DESCRIPTION
**Pull request by leogermani**
When the site is at a site-only connection, non administrators are not able to link their accounts to WordPress.com to enable features that require a user connection.

Editors, for example, can access the Settings page only to see Post By Email and Publicize, however they will not be able to link their accounts until and admin has done this before and enabled those features.

This PR hides the Settings page for non-admin while the site doesn't have a full connection

#### Changes proposed in this Pull Request:

* Hides the Settings menu for non admin while in site-only connection

#### Jetpack product discussion

p1619720622439100-slack-CBG1CP4EN?thread_ts=1619718679.433500&cid=CBG1CP4EN

#### Does this pull request change what data or activity we track or use?

No

#### Testing instructions:

* Connect at a site level only
* Check the Settings menu is available and working for admins
* Login as an editor and check the Settings menu is not available
* Complete the connection with a user
* Make sure Settings menu is working for both admins and editors

### CODE REVIEW

File: projects/plugins/jetpack/_inc/lib/admin-pages/class.jetpack-react-page.php [hunk 1]
```diff
@@ -65,9 +65,14 @@ function jetpack_add_dashboard_sub_nav_item() {
 	 * If user is allowed to see the Jetpack Admin, add Settings sub-link.
 	 *
 	 * @since 4.3.0
+	 * @since 9.7.0 If Connection does not have an owner, restrict it to admins
 	 */
 	function jetpack_add_settings_sub_nav_item() {
-		if ( ( ( new Status() )->is_offline_mode() || Jetpack::is_connection_ready() ) && current_user_can( 'edit_posts' ) ) {
+		if (
+			( ( new Status() )->is_offline_mode() || Jetpack::is_connection_ready() ) &&
+			current_user_can( 'edit_posts' ) &&
```
Comment (dereksmart) [inline projects/plugins/jetpack/_inc/lib/admin-pages/class.jetpack-react-page.php:73]: Would it make more sense to check against the `jetpack_connect` capability instead, and bail early? I'm just not 100% sure that `edit_posts` will catch them all.

**Comment by dereksmart** [comment]
This makes sense to me - I'm comfortable removing the settings page in this state for non-admins, since we're only showing settings for features that they can't use.

File: projects/plugins/jetpack/_inc/lib/admin-pages/class.jetpack-react-page.php (hunk 1 above)
Comment (leogermani) [reply projects/plugins/jetpack/_inc/lib/admin-pages/class.jetpack-react-page.php]: +1 using `jetpack_connect` cap. See the latest changes and tell me that's what you thought

**Review by dereksmart — APPROVED** [review]
It's working well for me. I would feel more comfortable if someone else also reviewed it cc @zinigor maybe?

**Review by jeherve — APPROVED** [review]
This tests well for me. Merging.

**Comment by jeherve** [comment]
Cherry-picked to `jetpack/branch-9.7` in 09a336ea95fcfd87ade6898ab76c145a1b7e4885

**Pull request merged**